core-default/src/main/java/me/aurium/beetle/defaults/task/sync/TimerSleepBehavior.java

```diff
@@ -0,0 +1,91 @@
+package me.aurium.beetle.defaults.task.sync;
+
+import me.aurium.beetle.api.task.futures.SyncTaskerFuture;
+import me.aurium.beetle.api.task.futures.TaskFuture;
+import me.aurium.beetle.api.task.sync.coordination.BlockingBehavior;
+import me.aurium.beetle.api.task.sync.coordination.SyncConstants;
+import me.aurium.beetle.api.task.sync.SyncPulser;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.LockSupport;
+
+public class TimerSleepBehavior implements BlockingBehavior {
+
+    private final int sleepTimer;
+
+    public TimerSleepBehavior(int sleepTimer) {
+        this.sleepTimer = sleepTimer;
+    }
+
+    public TimerSleepBehavior() {
+        this.sleepTimer = 200;
+    }
+
+    @Override
+    public <T> T waitForJoin(SyncTaskerFuture<T> future, SyncPulser pulser) {
+        while (true) {
+            pulser.pulse();
+
+            T result;
+            if (!(result = future.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
+                return result;
+            }
+
+            LockSupport.parkNanos(this, sleepTimer);
+        }
+    }
+
+    @Override
+    public <T> T waitForGet(SyncTaskerFuture<T> future, SyncPulser pulser) throws InterruptedException, ExecutionException {
+        while (true) {
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
+            }
+
+            pulser.pulse();
+
+            T result;
+            if (!(result = future.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
+                return result;
+            }
+
+            LockSupport.parkNanos(this, sleepTimer);
+        }
+    }
+
+    @Override
+    public <T> T waitForTimer(SyncTaskerFuture<T> future, SyncPulser pulser, TimeUnit waitType, long waitDuration) throws InterruptedException, TimeoutException {
+        long deadline = System.nanoTime() + waitType.toNanos(waitDuration);
+
+        while (true) {
+
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
+            }
+
+            pulser.pulse();
+
+            T result;
+            if (!(result = future.getNow(SyncConstants.CONSTANT())).equals(SyncConstants.CONSTANT())) {
+                return result;
+            }
+
+            //copy this shit from managedwaitstrategies because couldn't figure it out after a few hours of testing
```

Comment:
If you use unit tests you won't have to spend hours testing, and on top of that, you can keep the tests around and they'll run every time you make further changes

Reply:
Unit tests for sync? I mean unit tests in general is a good idea, but i have no idea how i'd test threaded with no mock-server (unless you mean creating a testing implementation of the coordinator in which case i have ~~no idea~~ a bad idea how i'd replicate a main-threaded environment

Reply:
To unit test using a "main-thread" I create a single-thread thread pool then run tasks on it